### PR TITLE
python api: optimizing dataframe creation from json responses

### PR DESCRIPTION
**Pull request by izcram**
also always returning seq_description field for consistency, adding tests

Resolves the awkward situation where the python client becomes a significant bottleneck in a query...

### CODE REVIEW

**Review by karasikov — APPROVED** [review]

File: metagraph/api/python/tests/data/search_with_align_response.json [hunk 1]
```diff
@@ -0,0 +1,6382 @@
...
+          "continent": "north_america",
+          "latitude": 40.706703186035156,
+          "longitude": -74.01113891601562,
+          "metasub_name": "CSD16-NYC-0923",
+          "num_reads": 17201082.0,
+          "sample_type": "environmental_microbiome",
+          "station": null,
+          "surface_material": "metal"
+        },
+        "sample": "haib17CEM4890_H2NYMCCXY_SL254785"
```
[the remark this answers is not in the document]
Comment (karasikov) [reply metagraph/api/python/tests/data/search_with_align_response.json:21]: Maybe we can compute a hash of the expected result? But I guess that will make it impossible to debug...

Do you see this json being used for debugging though, does it have more value than its hash?

File: metagraph/api/python/metagraph/helpers.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        return {**d, **props}
+
+    lst = [_build_dict(row, result) for result in json_res for row in
+           result['results']]
+
+    if lst:
+        return pd.DataFrame(lst)
+    else:
+        # columns may vary on the graph, so not adding column information
+        return pd.DataFrame()
```
[the remark this answers is not in the document]
Comment (izcram) [reply metagraph/api/python/metagraph/helpers.py:32]: yes, and yes. It can happen when no sequences were found.

File: metagraph/api/python/tests/data/search_with_align_response.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (izcram) [reply metagraph/api/python/tests/data/search_with_align_response.json:21]: I precisely would like to test the mechanism with the metadata. I'm not using it to check whether some result is correct, but to have an example to test whether the server responses are correctly converted to a dataframe. But yes, the file is a bit largish for what it is, I'll try have a smaller one next time

File: metagraph/api/python/tests/test_helpers.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import os
+from pathlib import Path
+import json
+from metagraph import helpers
+import pytest
+
+
+def _load_json_data(filename):
+    path = Path(os.path.abspath(__file__)).parent / 'data' / filename
```
[the remark this answers is not in the document]
Comment (izcram) [reply metagraph/api/python/tests/test_helpers.py:9]: Good point. Not sure there is this distinction in python. At least I would expect an automatic formatter for python would in general not be able to figure out the distinction between a numerical expression and a path concatenation. Above is the result of the formatter in pycharm. I guess, for a human reader it is clear from the context, that we are not dealing with an arithmetic expression.

File: metagraph/api/python/tests/test_helpers.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    with open(path, 'r') as f:
+        return json.load(f)
```
[the remark this answers is not in the document]
Comment (izcram) [reply metagraph/api/python/tests/test_helpers.py:12]: hmm, you mean for performance reasons? This is in the test_helpers and the tests run in like 25ms on my laptop, so I guess not worth trying to tune something here.

[the remark this answers is not in the document]
Comment (karasikov) [reply metagraph/api/python/tests/test_helpers.py:12]: Ah, of course. I didn't see that this is used only in tests.

File: metagraph/api/python/tests/data/search_with_align_response.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (karasikov) [reply metagraph/api/python/tests/data/search_with_align_response.json:21]: Yeah, then a smaller file can test the same thing, I guess. I guess this doesn't have edge cases anyway.
